src/user/user.service.ts

```diff
@@ -21,10 +24,28 @@ export class UserService {
         return this.cleanUser(createdUser);
     }
 
+    async findByLogin(UserDTO: LoginDTO) {
+        const { email, password } = UserDTO;
+        const user = await this.userModel.findOne({ email });
+        if (!user) {
```

Comment:
in UserDto you can use package Class-validator . this package have decartors  The Validate for us all fields that comes from A requst Object  .